src/gs-shell.c

```diff
@@ -782,6 +788,37 @@ gs_shell_reload_cb (GsPluginLoader *plugin_loader, GsShell *shell)
 	}
 }
 
+static void
+gs_shell_reload_and_select_removable_media_category (GsShell *shell,
+						     gboolean reload_overview,
+						     gboolean select_category,
+						     const gchar *category_name)
+{
+	/* using NULL for this below indicates it should not be selected */
+	const char *cat_name = (select_category ? category_name : NULL);
+
+	if (!reload_overview) {
+		reload_overview_and_select_category (shell, cat_name);
```

Comment:
This reads as: if not reload_overview, then reload overview and select category...
Is this correct?

Reply:
I noticed that earlier but apparently lost my note to address it.

It's carried directly over from this commit:
https://github.com/endlessm/gnome-software/commit/54863ceaf54cf5128f19b7f6e4338905a1b155d0#diff-b556be19f0ccc1af7d4666e07e9b7c04R2086

Do you know why that was written the way it was? I think I see a way to simplify it but I'm worried about somehow missing some important but subtle side effect.

As it is, I added a commit in my local branch to note the apparent confusion but haven't addressed it.

After the clean-up I mentioned in my last comment, this code is basically intact as it was in your commit I linked in this comment.